### ISSUE TITLE
Allow comments on API function pages

### ISSUE THREAD

**Issue by BWPanda** [backdrop-ops/api.backdropcms.org#82]
The Drupal API site allows you to post comments on function pages (e.g. https://api.drupal.org/api/drupal/modules%21system%21system.api.php/function/hook_permission/7.x#comment-28714). I often find these comments very useful and helpful in bug-fixing, finding the right function to use, wondering why a certain function no longer exists, etc.

Can we do something similar on the Backdrop API site? I imagine comments should be moderated so pages don't just fill up with support requests...

**Comment by klonos** [comment]
I support this as well; the d.org API comments have saved me hours of googling/troubleshooting countless times.

**Comment by docwilmot** [comment]
I have always thought that I proposed this issue years ago. Thanks for actually doing it. Agreed +++++++

**Comment by cellear** [comment]
Comments on doc pages can be really great.  Honestly, it was the high quality of the comments on the PHP and MySQL documentation pages that original drew me to the LAMP platform in the first place, and that is what led me to Drupal (and thence to Backdrop.)

_However_, the reason the MySQL and PHP pages jumped out at me is because I had previously suffered from many encounters with comments pages with very low signal-to-noise ratio.  It only takes one unfriendly commenter, or a handful of clueless ones, to make a board unusable.  I always assumed the PHP and MySQL pages had some strict moderation in play, but I didn't know the details.

The upshot is that I think if we open up the api site to comments we should give some thought to how we want to ensure quality.  Should it be moderated?  If so, who should (and who is willing) to take care of that?  Should be worry about succession?  Should we be prepared to turn it off if it doesn't achieve a certain level of success?  (I think sparsely-populated comment facilities can create an unsettling user experience.)

In short, how do we achieve high quality?

**Comment by BWPanda** [comment]
I'm all for moderation, and I'd be willing to do that. I see comments as being a place where you can provide information that'll help others (be it explaining something about the function better/a different way, pointing out 'gotchas', referencing other functions that can/should be used in conjunction, etc.). I don't see them as being a place where support should be asked for or provided (basically no questions allowed).

**Comment by BWPanda** [comment]
> basically no questions allowed

Although, I see benefit in allowing certain, common questions, as long as they're subsequently answered appropriately in order to assist people having the same issue.

**Comment by BWPanda** [comment]
Maybe we could implement something similar to PHP.net's guidleines for adding a note to their manual: https://www.php.net/manual/add-note.php

**Comment by BWPanda** [comment]
Here's an initial PR: https://github.com/backdrop-ops/api.backdropcms.org/pull/110

Comments are enabled on API content, but they could use some theming...
